flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -112,6 +114,13 @@ public TaskManagerRunner(
 		// Initialize the TM metrics
 		TaskExecutorMetricsInitializer.instantiateStatusMetrics(taskManagerMetricGroup, taskManagerServices.getNetworkEnvironment());
 
+		HeartbeatManagerImpl heartbeatManager = new HeartbeatManagerImpl(
+				taskManagerConfiguration.getTimeout().toMilliseconds(),
+				resourceID,
+				executor,
+				Executors.newSingleThreadScheduledExecutor(),
```

Comment:
I agree with that, and initially I want to reuse the executor in **RPCService** but it needs some modifications. I will introduce it in Runner in order to share among components.